### PR TITLE
🚸 Set highlight if clicked outside of word

### LINKED ISSUE #120: Playback position not moved when clicking left of text
When clicking left of the transcript text (but right of the speaker name column), the text-cursor is placed at the start of the transcript line. However the playback position is not moved to the same point.

https://user-images.githubusercontent.com/6545097/235660702-fbe00e12-8652-471d-9171-e5edf9804704.mov

### PR DESCRIPTION
**Pull request by pajowu**
Fixes #120

### CODE REVIEW

File: frontend/src/editor/transcription_editor.tsx [hunk 1]
```diff
@@ -48,7 +50,40 @@ function renderElement(
...
+                diff = clientX - bbox.right;
+              } else {
+                diff = 0;
+              }
+              if (diff < nearest_diff) {
+                nearest_diff = diff;
+                nearest = i;
+              }
+            }
+            window.dispatchEvent(new TextClickEvent(element.children[nearest]));
```
Comment (anuejn) [inline frontend/src/editor/transcription_editor.tsx:84]: can we also remove the handler in 109 following then?

Comment (pajowu) [reply frontend/src/editor/transcription_editor.tsx:84]: Nope, this doesn't work once you have multiline text. maybe we should just handle that cleanly here too and then get rid of 109

Comment (anuejn) [reply frontend/src/editor/transcription_editor.tsx:84]: I would prefer that (I also think, that this might be faster than having many listeners)

Comment (pajowu) [reply frontend/src/editor/transcription_editor.tsx]: I rewrote it to rather use the `selection` api and click the correct word

Comment (pajowu) [reply frontend/src/editor/transcription_editor.tsx]: Okay, after a bit more investigation: I think it's impossible to perfectly replicate the caret-behaviour in the onclick handler. I instead opted to use the `getSelection`-API in an onclick handler

**Review by anuejn — APPROVED** [review]
even if the code is a bit unholy: it works like a charm :)

**Pull request merged**